met-web/src/components/engagement/new/create/EngagmentVisibilityControl.tsx

```diff
@@ -0,0 +1,179 @@
+                            borderRadius: '4px',
+                            ml: 0,
+                            mt: 2,
+                            pb: 2,
+                            pr: 2,
+                            width: '100%',
+                        }}
+                    >
+                        <Grid item container spacing={1} alignItems="center">
+                            <Grid item>
+                                <BodyText bold color="primary.light">
+                                    Engagement URL
+                                </BodyText>
+                            </Grid>
+                            <When condition={isConfirmed}>
+                                <Grid item>
+                                    <IconButton
+                                        size="small"
+                                        onClick={() => setIsEditing(true)}
+                                        sx={{ display: 'inline-block' }}
+                                    >
+                                        <FontAwesomeIcon icon={faPencilAlt} />
+                                    </IconButton>
+                                </Grid>
+                            </When>
+                        </Grid>
+                        <Grid item>
+                            <BodyText>
+                                <span style={{ fontWeight: 'bold' }}>{siteUrl}/</span>
+                                {formSlug || '???'}
+                            </BodyText>
+                        </Grid>
+                        <When condition={!isConfirmed}>
+                            <Grid item container spacing={2} flexDirection="row" alignItems="center">
+                                <Grid item>
+                                    <Button variant="primary" onClick={() => setIsConfirmed(true)}>
+                                        Confirm
+                                    </Button>
+                                </Grid>
+                                <Grid item>
+                                    <Link
+                                        tabIndex={0}
+                                        sx={{ cursor: 'pointer', color: colors.type.regular.primary }}
+                                        onClick={() => {
+                                            setCurrentSlug(formSlug);
+                                            setIsEditing(true);
+                                        }}
+                                    >
+                                        Edit URL
+                                    </Link>
+                                </Grid>
+                            </Grid>
+                        </When>
+                    </Grid>
+                </When>
+
+                <When condition={isEditing}>
+                    <TextField
+                        value={currentSlug}
+                        onChange={setCurrentSlug}
+                        title="Engagement URL"
```

Comment:
It seems like maybe there's a precedent in this app for using the `title` attribute instead of having form labels? WCAG guidelines say we should go for labels whenever possible. Titles are somewhat acceptable but may be unreliable in terms of conveying field info to screen readers: https://www.w3.org/WAI/tutorials/forms/labels/#using-the-title-attribute

```suggestion
                        label="Engagement URL"
```